lib/core/sema/okl_sema_info.h

```diff
@@ -106,6 +106,7 @@ struct OklKernelInfo {
     explicit OklKernelInfo(const clang::FunctionDecl& decl)
         : decl(std::ref(decl)){};
     const std::reference_wrapper<const clang::FunctionDecl> decl;
-    std::list<OklLoopInfo> children = {};
+    std::list<OklLoopInfo*> topLevelOuterLoops = {};
```

Comment:
Storing a list of small objects (pointers) is not worth it.
I would suggest making a method that selects all the `Outer` loops for each loop branch in the kernel.

Reply:
Do you mean to create a method that calculates this list from `topLevelLoops`?

Reply:
Why not worth it?

Reply:
Regarding question. We have to store them somewhere. You start to parse from top to bottom. First outer must have parent a regular (if there is a regular before it), So this regular must be stored after parsing it, and retrieved when parsing first outer

Reply:
I see.
OK. I suggest to change `topLevelOuterLoops` to:
`std::vector<OklLoopInfo*> attributedChildren = {}`
Even if it is reallocated, we are storing raw pointers, so we are safe.

Reply:
> Do you mean to create a method that calculates this list from `topLevelLoops`?

Yes. possibly just recursively visit all the child, and select first attributed in each branch.

Reply:
Alright